plugins/wasm-cpp/extensions/key_auth/README_EN.md

```diff
@@ -0,0 +1,123 @@
+| `_match_route_`  | array of string | Optional，Optionally fill in one of `_match_route_`, `_match_domain_` | -      | Configure the route name to match                               |
+| `_match_domain_` | array of string | Optional，Optionally fill in one of `_match_route_`, `_match_domain_` | -      | Configure the domain name to match                                   |
+| `allow`          | array of string | Required                                                              | -      | For requests that meet the matching conditions, configure the name of the consumer that is allowed to access |
+
+**Warning：**
+- If the `_rules_` field is not configured, authentication will be enabled for all routes of the current gateway instance by default;
+- For a request that passes authentication, a `X-Mse-Consumer` field will be added to the request header to identify the name of the caller.
+
+# Example configuration
+
+## Enabled for specific routes or domains
+
+The following configuration will enable Key Auth authentication and authentication for gateway-specific routes or domain names. Note that the `credential` field cannot be repeated
+
+```yaml
+consumers:
+- credential: 2bda943c-ba2b-11ec-ba07-00163e1250b5
+  name: consumer1
+- credential: c8c8e9ca-558e-4a2d-bb62-e700dcc40e35
+  name: consumer2
+keys:
+- apikey
+in_query: true
+# Use the _rules_ field for fine-grained rule configuration
+_rules_:
+# Rule 1: Match by route name to take effect
+- _match_route_:
+  - route-a
+  - route-b
+  allow:
+  - consumer1
+# Rule 2: Take effect by domain name matching
+- _match_domain_:
+  - "*.example.com"
+  - test.com
+  allow:
+  - consumer2
+```
+
+The `route-a` and `route-b` specified in `_match_route_` in this example are the route names filled in when creating the gateway route. When these two routes are matched, calls whose `name` is `consumer1` will be allowed Access by callers, other callers are not allowed to access;
+
+`*.example.com` and `test.com` specified in `_match_domain_` in this example are used to match the domain name of the request. When the domain name matches, the caller whose `name` is `consumer2` will be allowed to access, and other calls access is not allowed.
+
+### Depending on this configuration, the following requests would allow access：
+
+Assume that the following request will match the route-a route
+
+**Set the API Key in the url parameter**
+```bash
+curl  http://xxx.hello.com/test?apikey=2bda943c-ba2b-11ec-ba07-00163e1250b5
+```
+**Set the API Key in the http request header**
+```bash
+curl  http://xxx.hello.com/test -H 'x-api-key: 2bda943c-ba2b-11ec-ba07-00163e1250b5'
+```
+
+After the authentication is passed, a `X-Mse-Consumer` field will be added to the header of the request. In this example, its value is `consumer1`, which is used to identify the name of the caller
```

Comment:
Before letter "X", we should use "an" instead of "a". And there should be a stop (".") at the end of sentence.